lib/spack/spack/cmd/style.py

```diff
@@ -202,13 +198,6 @@ def setup_parser(subparser):
         action="append",
         help="specify tools to skip (choose from %s)" % ", ".join(tool_names),
     )
-    subparser.add_argument(
-        "--spec-strings",
-        action="store_true",
-        help="upgrade spec strings in Python, JSON and YAML files for compatibility with Spack "
-        "v1.0 and v0.x. Example: spack style --spec-strings $(git ls-files). Note: this flag "
-        "will be removed in Spack v1.0.",
-    )
```

Comment:
We shouldn't remove this flag in 1.0 -- I think we should keep it so that people can still run it on their files.  There may be people with older files who want to update spec strings... I think it's still useful if you *know* your specs have variants after compilers. I also believe we can still parse this (it'll parse differently, but it's still valid and we can still move variants out of compiler deps to their parents).

Yes, running this on a more modern 1.0 files *may* break something but at least initially it's unlikely (as I don't think many people will have compiler specs with variants for a while). I think we can document how this is intended to be used and keep it for people who only encounter compiler deps after 1.0 is released.